docs/processing-build-steps.md

```diff
@@ -113,11 +113,28 @@ If a file has the `.twig` extension, it is first processed by the Twig renderer.
 previously collected build instructions are passed as template variables to the
 renderer.
 
+#### File conditions
+
 In the project template config, it is possible to provide a set of file conditions.
 Each file condition applies to a given path. The specified condition is then used
 in the step to decide whether to include or exclude the appropriate file. All other
 files will always be processed.
 
+#### Target file
+
+A file condition may also contain an additional `target` configuration. It can be
+used to configure a different target file location than the default location, e.g.
+if files with different contents should result in the same target file, depending
+on a given file condition.
+
+#### Multiple conditions
+
+It is possible to define more than one file condition for a single file. This is
+useful in combination with the configuration of a file target, e.g. if the file
+is expected on different locations for different versions of a configured framework.
+If more than one file condition evaluates to `true`, the first configured target is
+always used.
+
```

Comment:
Thanks, @eliashaeussler! Although the documentation itself is perfectly comprehensible I fear that we're starting to miss out on useful opportunities to showcase the powerful capabilities of the project-builder.

In this specific PR I reckon we offer a few more examples to have users grasps the underlying potential. I understand that this ask isn't an easy one because we should identify configuration examples that work for most developers despite their actual project. 

Now, I understand that the project-builder is designed to offer a very generic approach of kickstarting composer-based  projects. However, the actual implementation requires a template, of course. At this very moment I only see TYPO3 and Symfony implementations making use of the project builder. This might hopefully change in the future.

**TL;DR;**
Please add an example. Similiar to the PR description, perhaps.

Reply:
Since the documentation is bigger issue we'll take that into account here #50.